### PR TITLE
Timestamp standardization

### PR DESCRIPTION
**Pull request by maackle**
### INCLUDED CHANGES:
- Many instances of raw integers representing timestamps have been converted to use `Timestamp`
- Most database columns for timestamps are in microseconds.
    - Exceptions include AgentInfo columns which still use milliseconds.

### TODO:
- [x] CHANGELOG(s) updated with appropriate info

### CODE REVIEW

File: crates/holochain_p2p/src/spawn/actor.rs [hunk 1]
```diff
@@ -301,10 +301,10 @@ impl WrapEvtSender {
         &self,
         dna_hash: DnaHash,
         agents: Vec<(AgentPubKey, kitsune_p2p::dht_arc::DhtArcSet)>,
-        window_ms: TimeWindowMs,
+        window_ms: TimeRange,
```
Comment (neonphog) [inline crates/holochain_p2p/src/spawn/actor.rs:304]:
```suggestion
        window: TimeRange,
```

File: crates/holochain_p2p/src/types/event.rs [hunk 2]
```diff
@@ -216,10 +216,10 @@ ghost_actor::ghost_chan! {
         fn query_op_hashes(
             dna_hash: DnaHash,
             to_agents: Vec<(AgentPubKey, kitsune_p2p::dht_arc::DhtArcSet)>,
-            window_ms: TimeWindowMs,
+            window_ms: TimeRange,
```
Comment (neonphog) [inline crates/holochain_p2p/src/types/event.rs:219]:
```suggestion
            window: TimeRange,
```

File: crates/kitsune_p2p/direct/src/persist_mem.rs [hunk 3]
```diff
@@ -486,7 +486,7 @@ impl AsKdPersist for PersistMem {
         &self,
         root: KdHash,
         agent: KdHash,
-        _window_ms: TimeWindowMs,
+        _window_ms: TimeRange,
```
Comment (neonphog) [inline crates/kitsune_p2p/direct/src/persist_mem.rs:489]:
```suggestion
        _window: TimeRange,
```

File: crates/kitsune_p2p/direct/src/types/persist.rs [hunk 4]
```diff
@@ -85,7 +85,7 @@ pub trait AsKdPersist: 'static + Send + Sync {
         &self,
         root: KdHash,
         agent: KdHash,
-        window_ms: TimeWindowMs,
+        window_ms: TimeRange,
```
Comment (neonphog) [inline crates/kitsune_p2p/direct/src/types/persist.rs:88]:
```suggestion
        window: TimeRange,
```

File: crates/kitsune_p2p/kitsune_p2p/src/types/event.rs [hunk 5]
```diff
@@ -14,7 +15,7 @@ pub struct QueryOpHashesEvt {
     /// The agents from which to fetch, along with a DhtArcSet to filter by.
     pub agents: Vec<(KAgent, DhtArcSet)>,
     /// The time window to search within.
-    pub window_ms: TimeWindowMs,
+    pub window_ms: TimeRange,
```
Comment (neonphog) [inline crates/kitsune_p2p/kitsune_p2p/src/types/event.rs:18]:
```suggestion
    pub window: TimeRange,
```

File: crates/kitsune_p2p/timestamp/src/lib.rs [hunk 6]
```diff
@@ -315,28 +277,28 @@ impl Timestamp {
...
-        Self(secs, nanos)
+    /// Subtract a duration, clamping to MIN if overflow
+    pub fn saturating_sub(&self, rhs: &core::time::Duration) -> Timestamp {
+        self.checked_sub(rhs).unwrap_or(Self::MIN)
+    }
+
+    /// Convert this timestamp to fit into a SQLite integer which is an i64.
+    /// The value will be clamped to the valid range supported by SQLite
+    pub fn into_sql_lossy(self) -> Self {
+        Self(i64::clamp(self.0, -62167219200 * MM, 106751991167 * MM))
```
Comment (neonphog) [inline crates/kitsune_p2p/timestamp/src/lib.rs:301]: something's not right with this comment / logic... obviously sqlite can store the full range of i64 values.

**Review by neonphog — APPROVED** [review]
other than some _ms holdovers and one question about a comment, lgtm : )

File: crates/kitsune_p2p/timestamp/src/lib.rs (hunk 6 above)
Comment (maackle) [reply crates/kitsune_p2p/timestamp/src/lib.rs:301]: @neonphog but I believe that SQL's datetime-related functions only operate on i64s within this range.

**Review by freesig — APPROVED** [review]
LGTM

**Pull request merged**